R/tdata.R

```diff
@@ -0,0 +1,188 @@
+#' Create a `tdata` Object
+#'
+#' Create a new object called `tdata` which contains `data`, a `reactive` list of data.frames
+#' (or `MultiAssayExperiment`), with attributes:
+#' \itemize{
+#'   \item{`code` (`reactive`) containing code used to generate the data}
+#'   \item{join_keys (`JoinKeys`) containing the relationships between the data}
+#'   \item{metadata (`named list`) containing any metadata associated with the data frames}
+#' }
+#' @name tdata
+#' @param data A `named list` of `data.frames` (or `MultiAssayExperiment`)
+#'  which optionally can be `reactive`.
+#'   Inside this object all of these items will be made `reactive`.
+#' @param code A `character` (or `reactive` which evaluates to a `character`) containing
+#'   the code used to generate the data. This should be `reactive` if the code is changing
+#'   during a reactive context (e.g. if filtering changes the code). Inside this
+#'   object `code` will be made reactive
+#' @param join_keys A `teal.data::JoinKeys` object containing relationships between the
+#'   datasets.
+#' @param metadata A `named list` each element contains a list of metadata about the named data.frame
+#' Each element of these list should be atomic and length one.
+#' @param check (`logical`) value whether reproducibility check is requested or not.
+#' @param label named (`list`) of datasets labels.
+#'
+#' @return A `tdata` object
+#' @examples
+#'
+#' data <- new_tdata(
+#'   data = list(iris = iris, mtcars = reactive(mtcars), ds1 = data.frame(x = 1:10)),
+#'   code = "iris <- iris
+#'     mtcars <- mtcars
+#'     dd <- data.frame(x = 1:10)",
+#'   metadata = list(ds1 = list(author = "NEST"), iris = list(version = 1)),
+#'   check = TRUE,
+#'   label = list(iris = "iris", mtcars = "mtcars", ds1 = "ds1")
+#' )
+#'
+#' # Extract a data.frame
+#' isolate(data[["iris"]]())
+#'
+#' # Get code
+#' isolate(get_code_tdata(data))
+#'
+#' # Get metadata
+#' get_metadata(data, "iris")
+#'
+#' @export
+new_tdata <- function(data, code = "", join_keys = NULL, metadata = NULL, check = FALSE, label) {
```

Comment:
the `check` argument here is not in its perfect place and is only passed downstream inside `FilteredData` to be called in `teal::get_datasets_code` to evaluate if an warning message should be returned in the SRC or not.
An alternative to this would be not to pass `check` at all but rather to evaluate if `check` is TRUE or FALSE upstream and throw a warning message in the console. This will not return a warnings message in the SRC as it is done now but will avoid us the passing an unnecessary argument all through the teal app setup.
@nikolas-burkoff @donyunardi @lcd2yyz

Reply:
https://github.com/insightsengineering/teal.data/issues/97